firmware/library/L1_Drivers/ssp.hpp

```diff
@@ -70,3 +71,235 @@ class SspInterface
+            ssp_registers[pssp]->CR0 |=
+                (clock_polarity_ << kPolarityBit) | (clock_phase_ << kPhaseBit);
+            ssp_registers[pssp]->CR0 |= (clock_divider_ << kDividerBit);
+            // enable SSP
+            ssp_registers[pssp]->CR1 &= ~(1 << 1);
+            ssp_registers[pssp]->CR1 |= (1 << 1);
+            ssp_registers[pssp]->CR1 &= ~(1 << kMasterModeBit);
+            ssp_registers[pssp]->CR1 |= (master_mode_ << kMasterModeBit);
+            peripheral_initialized = 1;
+        }
+        return peripheral_initialized;
+    }
+
+    // Renamed from get_status()
+    // Checks if the SSP controller is idle.
+    // @return 1 - the controller is sending or receiving a data frame.
+    // @return 0 - the controller is idle.
+    bool GetTransferStatus() override
+    {
+        return (ssp_registers[static_cast<uint32_t>(pssp_)]->SR &
+            (1 << kDataLineIdleBit));
+    }
+
+    // Transfers a data frame to an external device using the SSP
+    // data register. This functions for both transmitting and
+    // receiving data. It is recommended this region be protected
+    // by a mutex.
+    // @param data - information to be placed in data register
+    // @return - received data from external device
+    uint16_t Transfer(uint16_t data) override
+    {
+        uint32_t pssp = static_cast<uint32_t>(pssp_);
+
+        ssp_registers[pssp]->DR = data;
+        while (GetTransferStatus()) { continue; }
+        return static_cast<uint16_t>(ssp_registers[pssp]->DR);
+    }
+
+    // Set the PCONP bits to power on peripheral
+    // uses the pssp_ set in constructor
+    void SetPeripheralPower() override
+    {
+        pconp_bits_ =
+            (pssp_ == Peripheral::kSsp0) ? kPconp0 : kPconp1;
+    }
+
+    // Sets the various modes for the Peripheral
+    // @param mode - master or slave mode
+    // @param frame - format for Peripheral data to use
+    // @param size - number of bits per frame
+    void SetPeripheralMode(MasterSlaveMode mode,
+                            FrameMode frame,
+                            DataSize size) override
+    {
+        master_mode_ = mode;
+        frame_format_ = frame;
+        data_size_ = size;
+    }
+
+    // Sets the clock rate for the Peripheral
+    void SetClock(bool polarity,
+                    bool phase,
+                    uint8_t divider,
+                    uint8_t prescaler) override
+    {
+        clock_polarity_ = polarity;
+        clock_phase_ = phase;
+        clock_divider_ = divider;
+        clock_prescaler_ = prescaler;
+    }
+
+    ~Ssp()
+    {
+        // do nothing
+    }
+
+ private:
+    PinConfigureInterface & mosi_;
+    PinConfigureInterface & miso_;
+    PinConfigureInterface & sck_;
+
+    PinConfigure mosi_pin_;
+    PinConfigure miso_pin_;
+    PinConfigure sck_pin_;
+
+    // SSP member variables
+    Peripheral pssp_;                // SSP interfaces
+
+// TODO(Khrysta): Replace following variables with get/set functions

```

Comment:
Indent this, also, create a new issue and put the issue number in the (). For example TODO(#1234) where 1234 is the issue number. This is so, anyone else can pick up where you left off rather than going and finding you to fix this problem.